### PR TITLE
Multiple attacks

### PR DESCRIPTION
**Pull request by TerrenceJChan**
Allows the user to select from a list of attacks. Commands now also accept multiple arguments.

### CODE REVIEW

File: src/utils/optionsMessage.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+  * @param {object} player Contains the player's information
+  * @returns {string} A message to display detailing possible user actions
+  */
+export const optionsMessage = ({ player }) => {
+  const skillDescriptions = () => {
+    let listOptions = '';
+    for (let i = 0; i < 3; i++) {
```
Comment (foxfriends) [inline src/utils/optionsMessage.js]:
```suggestion
    for (let i = 0; i < player.skills.length; i++) {
```

File: src/index.js [hunk 2]
```diff
@@ -92,3 +89,5 @@ app.run(bot, () => {
 });
 
 // msg.reply("noot noot");
+
+// msg.reply("noot noot");
```
Comment (foxfriends) [inline src/index.js]:
```suggestion
```

File: src/index.js [hunk 3]
```diff
@@ -59,28 +56,28 @@ app.use((msg) => {
   }
   messageAt = Date.now();
   // "Removes" the COMMAND_PREFIX from the user's message
-  const command = msg.content.slice(COMMAND_PREFIX.length);
+  const command = msg.content.slice(COMMAND_PREFIX.length).split(' ');
```
Comment (foxfriends) [inline src/index.js]: Can destructure those two, then they have more meaningful names that `command[0]` and `command[1]`
```suggestion
  const [command, arg] = msg.content.slice(COMMAND_PREFIX.length).split(' ');
```

File: src/index.js [hunk 4]
```diff
@@ -59,28 +56,28 @@ app.use((msg) => {
...
   // Rejects invalid command
-  if (!ALL_COMMANDS.has(command)) {
+  if (!ALL_COMMANDS.has(command[0])) {
```
Comment (foxfriends) [inline src/index.js]:
```suggestion
  if (!ALL_COMMANDS.has(command)) {
```

File: src/index.js [hunk 5]
```diff
@@ -59,28 +56,28 @@ app.use((msg) => {
...
     msg.channel.send(`Unknown command: ${command}`);
     return;
   }
   // Handles valid command
   let commandHandler;
-  if (ctx.enemy) {
-    commandHandler = FIGHT_COMMANDS[command];
+  if (ctx.enemy && ctx.player) {
+    commandHandler = FIGHT_COMMANDS[command[0]];
```
Comment (foxfriends) [inline src/index.js]:
```suggestion
    commandHandler = FIGHT_COMMANDS[command];
```

File: src/index.js [hunk 6]
```diff
@@ -59,28 +56,28 @@ app.use((msg) => {
...
     if (!commandHandler) {
       msg.channel.send('This command is not available during battle!');
       return;
     }
   } else {
-    commandHandler = TOWN_COMMANDS[command];
+    commandHandler = TOWN_COMMANDS[command[0]];
```
Comment (foxfriends) [inline src/index.js]:
```suggestion
    commandHandler = TOWN_COMMANDS[command];
```

File: src/index.js [hunk 7]
```diff
@@ -59,28 +56,28 @@ app.use((msg) => {
...
     if (!commandHandler) {
       msg.channel.send('This command is only available during battle!');
       return;
     }
   }
-  const response = commandHandler(ctx);
+  const response = commandHandler(ctx, command[1]);
```
Comment (foxfriends) [inline src/index.js]:
```suggestion
  const response = commandHandler(ctx, arg);
```

File: src/commands/attack.js [hunk 8]
```diff
@@ -13,9 +14,28 @@ const addItem = (drop, inventory) => {
...
-export const attack = (ctx) => {
+export const attack = (ctx, option) => {
   const { player, inventory, enemy } = ctx;
-  let enemyDamage = Math.floor((player.atk * random20()) - (enemy.def * random20()));
+  // Converts second argument into a number
+  if (!isNaN(option)) {
+    option = parseInt(option);
+  }
+  // We do not use !attack 0, so we do some math to change this. Also checks for valid input
+  if (Number.isInteger(option) && option > 0 && option < 4) {
```
Comment (foxfriends) [inline src/commands/attack.js]:
```suggestion
  if (Number.isInteger(option) && option > 0 && option <= player.skills.length) {
```

File: player/Player.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+export default class Player {
+  constructor() {
+    this.name = 'Robert';
+    this.hp = 100;
+    this.atk = 20;
+    this.def = 7;
+    this.skills;
```
Comment (foxfriends) [inline player/Player.js]: This line does nothing without assigning something
```suggestion
    this.skills = [];
```

File: src/index.js (hunk 2 above)
Comment (TerrenceJChan) [reply src/index.js]: :(

**Review by foxfriends — APPROVED** [review]

**Pull request merged**